api/models.go

```diff
@@ -218,11 +220,31 @@ type Transaction struct {
 	CreatedAt       string      `json:"createdAt"`
 	SettledAt       *string     `json:"settledAt"`
 	AppId           *uint       `json:"appId"`
-	Metadata        Metadata    `json:"metadata,omitempty"`
+	Metadata        *Metadata   `json:"metadata,omitempty"`
 	Boostagram      *Boostagram `json:"boostagram,omitempty"`
 }
 
-type Metadata = map[string]interface{}
+type Metadata struct {
+	Comment   string      `json:"comment,omitempty"`
+	Nostr     *NostrEvent `json:"nostr,omitempty"`
+	PayerData *PayerData  `json:"payerData,omitempty"`
+}
+
+type NostrEvent struct {
+	Content   string     `json:"content"`
+	CreatedAt int64      `json:"createdAt"`
+	ID        string     `json:"id"`
+	Kind      int        `json:"kind"`
+	PubKey    string     `json:"pubkey"`
+	Sig       string     `json:"sig"`
+	Tags      [][]string `json:"tags"`
+}
+
+type PayerData struct {
```

Comment:
PayerData can also contain anything. We just know about these fields

Reply:
Same as above^